src/main/java/org/springframework/data/couchbase/core/mapping/event/ReactiveAuditingEntityCallback.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.couchbase.core.mapping.event;
+
+import org.reactivestreams.Publisher;
+import org.springframework.beans.factory.ObjectFactory;
+import org.springframework.core.Ordered;
+import org.springframework.data.auditing.AuditingHandler;
+import org.springframework.data.auditing.ReactiveIsNewAwareAuditingHandler;
+import org.springframework.data.mapping.callback.EntityCallback;
+import org.springframework.data.mapping.context.MappingContext;
+import org.springframework.util.Assert;
+
+/**
+ * Reactive {@link EntityCallback} to populate auditing related fields on an entity about to be
+ * saved. Based on <a href="https://github.com/spring-projects/spring-data-mongodb/blob/3.1.6/spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/mapping/event/ReactiveAfterConvertCallback.java">ReactiveAfterConvertCallback</a>
+ *
+ * @author Jorge Rodríguez Martín
+ * @since 4.2
+ */
+public class ReactiveAuditingEntityCallback implements ReactiveBeforeConvertCallback<Object>,
```

Comment:
Make this also implement ReactiveAfterConvertCallback<Object>.  Add onAfterConvert(Object, CouchbaseDocument, String)

Reply:
I think it is not correct to add `ReactiveAfterConvertCallback`. Audit event should only be associated to a single event (ReactiveBeforeConvertCallback).

If we also add the `ReactiveAfterConvertCallback`, in each conversion the information of the fields annotated with @CreatedBy, @CreatedDate, @LastModifiedBy and @LastModifiedDate will be modified 2 times and the returned object will have different values for the date type fields than the saved CouchbaseDocument.